### PR TITLE
Fix broken postgresql connection

### PR DESCRIPTION
**Pull request by thomastvedt**
See issue #218 

This PR adds max connection lifetime to postgresql db connection. Hopefully this will help with our issue where the auth plugin / mosquitto stops responding after a while (typically 30+ days..)

We're testing this version now in production, we'll test it for a while to verify that it helps on stability. => seems to be working 

https://github.com/iegomez/mosquitto-go-auth/issues/218
https://stackoverflow.com/questions/50338338/intermittent-connection-reset-by-peer-sql-postgres

I also bumped  
from: github.com/jmoiron/sqlx v1.2.0   
to: github.com/jmoiron/sqlx v1.3.4

### CODE REVIEW

**Comment by iegomez** [comment]
Hey, @thomastvedt!

Just reaching out since we're getting close to the time frame you mentioned when Postgres starts becoming unresponsive. Ping me when you get some results from your tests and we'll merge it accordingly.

**Comment by thomastvedt** [comment]
> Hey, @thomastvedt!
> 
> Just reaching out since we're getting close to the time frame you mentioned when Postgres starts becoming unresponsive. Ping me when you get some results from your tests and we'll merge it accordingly.

Hi @iegomez , seems to be working fine, even with longer periods of inactivity (holidays) 👍. I'll delete the temporary build script / clean up, and you can merge :)

File: backends/db.go [hunk 1]
```diff
@@ -41,5 +42,7 @@ func OpenDatabase(dsn, engine string, tries int) (*sqlx.DB, error) {
 		return nil, fmt.Errorf("couldn't ping database %s: %s", engine, err)
 	}
 
+	db.SetConnMaxLifetime(time.Minute)
```
Comment (iegomez) [inline backends/db.go:45]: Maybe we should make this an option so users may tune it. What do you think?

File: backends/db.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ func OpenDatabase(dsn, engine string, tries int) (*sqlx.DB, error) {
 		return nil, errors.Wrap(err, "database connection error")
 	}
 
+
```
Comment (iegomez) [inline backends/db.go:22]: ?

**Review by iegomez — COMMENTED** [review]
Just left a couple of comments before we merge.

File: backends/db.go (hunk 2 above)
Comment (thomastvedt) [reply backends/db.go]: Removed 👍🏻

File: backends/db.go (hunk 1 above)
Comment (thomastvedt) [reply backends/db.go:44]: Hey, that could be a good idea? Unfortunately I'm not a Go expert, and I'm not sure how to implement that :) Do you want to have a look at it?

Comment (iegomez) [reply backends/db.go:44]: @thomastvedt sorry, I was on vacation and just got back. Sure, I'll do so.

Comment (thomastvedt) [reply backends/db.go:44]: Great, thanks 😃

Comment (iegomez) [reply backends/db.go:44]: Terribly sorry for the delay, but here it is, @thomastvedt: https://github.com/iegomez/mosquitto-go-auth/pull/224

**Comment by thomastvedt** [comment]
Closing this, replaced by #224

**Pull request closed without merging**